### DISCUSSION TITLE
Use Cases for Deskreen

### DISCUSSION THREAD

**Discussion by pavlobu** [pavlobu/deskreen#48] [Show and tell]
Hi all. I created this thread where you can share your use cases for Deskreen. If you think you have something interesting to say about how you use Deskreen, please don't hesitate and leave your message here. Sharing your experience will benefit the others.

**Comment by pavlobu** [comment]
A [user](https://www.reddit.com/user/suby) from reddit reported using Deskreen with [barrier](https://github.com/debauchee/barrier) 

[read on reddit](https://www.reddit.com/r/linux/comments/l3etrn/i_created_deskreen_this_is_a_desktop_app_that/gkqf4ul?utm_source=share&utm_medium=web2x&context=3)

**Comment by pavlobu** [comment]
A user from reddit [reported](https://www.reddit.com/r/linux/comments/l3etrn/i_created_deskreen_this_is_a_desktop_app_that/gkw8dmm?utm_medium=android_app&utm_source=share&context=3)  using Deskreen to make a windows laptop a secondary screen for linux desktop. In combination with: [https://github.com/dianariyanto/virtual-display-linux](https://github.com/dianariyanto/virtual-display-linux)
Took 5 min to setup and all running smoothly

**Reply by aaylnx** [reply]
I'm VDL also. Just downloaded and ran it yesterday.  It was super easy to set up and is working fine. I tried it out after I ordered a VGA dummy plug for my Lenovo T440p off of Amazon. Looks like I won't be needing that now.

Thanks for your work on this project. It is great!

**Reply by pavlobu** [reply]
You're welcome! Enjoy!

**Reply by mario-tux** [reply]
I'm wondering if it works with Wayland (I mean VDL)

**Comment by RafaelLinux** [comment]
Tip for openSUSE 15.2 / Fedora / RedHat users  with firewall enabled, using Deskcreen Appimage version.
We need to open port 3131, so we need to open it and restart firewall:
```
su
firewall-cmd --zone=public --permanent --add-port=3131/tcp
systemctl restart firewalld
exit
```
That's all

**Comment by ericsmalling** [comment]
I'm not a user yet (just learned of Deskreen today) but I currently use a competing, commercial tool that does similar functionality.  My use case is to mirror my screen to an iPhone that is mounted in a small teleprompter on the front of my DSLR (Parrot 2).  When doing zoom meetings I use this so I'm reminded to look at the lens instead of my monitor.

The deficiencies with my current setup are:

1. The image is reversed since the teleprompter has me looking at a mirror of the phone screen. It would be nice if there was a mode that put the remote screen into a mirror-image mode. (horizontally flipped)
2. The app on the phone does not have an orientation lock, thus every time I start it up it I have to remove the phone from the teleprompter mount, tilt it to landscape, replace phone in mount.  (It sits flat, parallel to the floor so the phone doesn't "know" what orientation to use and defaults to portrait.)
3. It would be awesome if I could mirror a single window instead of an entire display.

I have previously asked the company that makes this app if they could add these features and got a "we'll add them to the list" type response.   If Deskreen had my bullet 1 and a way to do bullet 2, I'd probably switch to it.  If it had all three that would be awesome!

Thanks, hope this input is valuable to the team.

**Reply by pavlobu** [reply]
Hi, interesting use case.

Is it something related to this feature? https://github.com/pavlobu/deskreen/issues/93
I was thinking @tobi was talking about a mirror screen feature, for me mirror screen means "cast entire screen" which Deskreen already has.

But here I see you want something like flipping a screen vertically or rotating a video.
This is really simple to add. And I have found a solution on stackoverflow:
https://stackoverflow.com/questions/14455844/is-that-possible-to-make-video-mirrored

I'll be looking on how to add this functionality 👌 Stay tuned

**Reply by pavlobu** [reply]
do you happen to know a good JS library for manipulating video stream frames for simple rotating or flipping them? 
coz now if I simply flip video element with css it will be flipped in normal mode, but it will not be flipped in default video player full screen mode.

**Reply by pavlobu** [reply]
simple flip on CSS property can be added with no problem, but it will not be working in default browser mode

rotation function is more complicated, for that will need to change a code a lot and it will take time.

**Reply by pavlobu** [reply]
Regarding 2. Have you tried Deskreen? It should be working ok with Orientation in web browser

**Reply by ericsmalling** [reply]
I've not tried Deskreen yet but will soon.   Unless the browser has a way to lock orientation, I'm not sure if there is going to be a way to accomplish item 2 as iOS on iPhone doesn't allow it globally (that I know of).

FWIW - I may have found a way to do this: https://www.techwalla.com/articles/how-to-lock-your-iphone-in-portrait-or-landscape

**Reply by ericsmalling** [reply]
> Hi, interesting use case.
> 
> Is it something related to this feature? #93
> I was thinking @tobi was talking about a mirror screen feature, for me mirror screen means "cast entire screen" which Deskreen already has.
> 
> But here I see you want something like flipping a screen vertically or rotating a video.
> This is really simple to add. And I have found a solution on stackoverflow:
> https://stackoverflow.com/questions/14455844/is-that-possible-to-make-video-mirrored
> 
> I'll be looking on how to add this functionality 👌 Stay tuned

Yes, that's it.  From the example in that StackOverflow thread, it looks like this would do the trick:

```
{
    transform: rotateY(180deg);
    -webkit-transform:rotateY(180deg); /* Safari and Chrome */
    -moz-transform:rotateY(180deg); /* Firefox */
}
```

**Reply by tobi** [reply]
Yes, sorry I should have been more clear in my ticket. 

I think implementing this in CSS for starters is fine. It would be amazing to get a wizz-bang WASM frame flipper for fullscreen support, but not strictly needed for this use case. 

Once you get this feature in, I'd absolutely think you should put it on the main site. There are a lot of people who are frustrated by lack of good solution in the corporate setting.  Currently the best solution that I know of involves a hardware dongle ( https://www.youtube.com/watch?v=swiAke-SGD0&feature=emb_title )

**Reply by pavlobu** [reply]
@tobi @ericsmalling 
alright! Flip button added in new release v1.0.4 🎉
https://github.com/pavlobu/deskreen/releases/tag/v1.0.4

it is displayed when you click Gear icon button, the one that is next to Play button

it is made with CSS for now, so you need to use a native browser for your tablet or smartphone to use a featured video player that supports flipping video in full screen mode. If you have IPhone or IPad use Safari. If you are using an Android device, use Chrome. When using a proper browser, you will be able to use a featured video player that supports flipping video in full screen mode.

**Reply by pavlobu** [reply]
@ericsmalling
so v1.0.4 solves 1) and 3) of your list

if you switch to Deskreen and if you are a blogger, consider letting your friends and subscribers know about Deskreen ;)

Cheers,
Paul

**Reply by pavlobu** [reply]
@tobi
Thanks for advice! 👍 On website I have added a teleprompter Flip Screen Mode into Features section

**Comment by ZhangTianrong** [comment]
Interesting use case: I wrote an iOS shortcut that inject some simple js to the webpage to duplicate the stream. Now I can wear a VR headset and use my computer while I am lying in bed either on my back or on the side (Most people use VR headset so they can use the gyro in the phone to navigate but I am just so lazy and I want to browse webpages in bed without holding my phone and there is literally no VR browsers for iOS anymore in AppStore. They require you to sit still anyway).

```javascript
var itm = document.querySelector("#react-player-wrapper-id > video");
itm.style.width="41.25%";
var cln = itm.cloneNode(true);
// You might want to tweak around this attribute a bit to adjust the reduced image distance according to your preferences
cln.style.marginLeft="2.5%"
document.querySelector("#react-player-wrapper-id").appendChild(cln);
cln.srcObject=itm.srcObject;
document.querySelector("#react-player-wrapper-id").style.textAlign="center";
```

The only thing next I guess is to find a way to use the keyboard more easily. Speech recognition might be helpful but I need to enter non-English characters at the same time.

**Reply by pavlobu** [reply]
Hey @ZhangTianrong  this is an interesting use case :) You can submit a PR with your solution, and maybe in UI on client side we can add an extra button in Video Settings along with quality change buttons and Flip Screen button that already exists in UI. And give it an Icon with VR googles and call it "VR". By clicking that button it will toggle a VR mode on and off
icons for VR can be found here for example:
https://iconify.design/icon-sets/?query=vr

I'm sure this feature may help not only lazy people but also some people with limited abilities. Sound like a really useful stuff !

**Comment by justinkelly** [comment]
My current use case:

_At work - no admin access, no docks, no hdmi cables, no extra monitors_

Using deskreen portable for windows + dummy hdmi plug
* on a work provided laptop (with no admin rights)
* I can use any work PC (monitor + PC integrated so no hdmi available) (again no admin rights) as a second screen
* very , very useful and with the hdmi dummy plug - it just works!

**Reply by Jetol** [reply]
Hello，justinkelly How to  PC integrated

**Comment by DavidGP** [comment]
Hello all,

Thanks very much for the awesome tool Deskreen. 

I wonder if it is possible to set up a virtual screen that has a higher resolution than any actual screen of any present device. The objective would be to create screenshots of applications with a higher resolution (i.e. larger pixelwise size) than the physical resolution of the monitor. For example, create 8K resoution screenshots, and/or screenshots in rotated/portrait mode.

**Comment by luigieai** [comment]
In my desk i have my personal machine and my work machine, my personal machine is a CPU and work's a notebook. When i'm working i like to have the same mouse and keyboard, and in my personal CPU i have dual monitor, so i use deskreen in second montitor with barrier, a software that act like a KVM Switch.
It's already 4 months that i'm using deskreen as daily driver and i cannot be more happier :)

**Comment by DominikNovosel** [comment]
When writing with voice commands using serenade.ai, i use Deskreen to display its window on my phone, keeping it from taking up space on my primary display.

**Comment by positive-medium** [comment]
So I am trying to use a Facebook Portal+ device as a second screen using Deskreen.  This works great if the screen on the device is in landscape mode, but I'm trying to make it work in portrait (vertical) mode.  If I try to use it that way, it shrinks the display so that it only takes up the top part of the Portal+.  I tried rotating the virtual display in MacOS but then it only shows part of the display on the Portal+.  Unfortunately I can't modify the CSS or anything like that on the Portal+'s included browser.  Is there any way I could get this working?

**Comment by lipner** [comment]
Trying to use a Tesla vehicle's screen as a second screen for my Macbook / as a primary screen for a Raspberry Pi.

**Comment by mburgess00** [comment]
Deskreen looks super slick, but I'd really love to have input. I do a lot of CAD and drawing work on Linux and this could replace something like a Wacom tablet.

**Comment by ghost** [comment]
is it possible to use deskscreen as an imput device

**Reply by samggreenberg** [reply]
Not at the moment, no.

**Comment by beijoca** [comment]
I work in education and am somewhat forced into the apple eco-system, as many classrooms are equipped with apple-tvs and some rooms no longer have HDMI access. Recently I finally moved from a macbook to a linux-laptop as my main machine, as a really want to get out of the apple ecosystem, and while looking for a way to connect linux via airplay I came across Deskreen. Now I can pick-up any iPad  at work, hook it up to apple-TV and mirror my Note-Taking-Application from my laptop to the iPad, so this is a nice use case for Deskreen.

One Question: is it possible to forward sound as well? I would love to also be able to play videos on my laptop and forward these to appletv in the above mentioned way. thanks in advance

**Comment by samggreenberg** [comment]
I think Deskreen is the only way to expand a display *from* a chromebook. (Lots of ways to expand *to*, of course.)
If so, that's a heck of a use-case.

**Reply by jamesshannon** [reply]
@samggreenberg Sounds like you're just assuming it'll work from chromeos? If you've actually gotten this to work then could you share how?

**Comment by EDIflyer** [comment]
Just found Deskreen and absolutely love it. Also just tried it out on my Raspberry Pi which is at another location connected via Wireguard VPN - still streams very nicely, great for monitoring server stats, etc.

**Comment by JonB256** [comment]
Just installed and first use was for Karaoke. Each singer has their own tablet. Worked perfectly!

**Comment by skunkmommy** [comment]
I discovered [Weylus](https://github.com/H-M-H/Weylus) which lets you use something like a phone as a crappy graphics tablet. It has a second screen feature, but sadly it's only on linux.
So i used this in conjunction with IddDisplayDriver and now i can use my phone as a third monitor to move my youtube vids to, or pretty much anything.
I could even use my Steam Deck as a fourth monitor on my pc, allowing a wireless dual-monitor solution that's still running on my main PC.
This seems really cool, thx for making it a thing

**Comment by SeanRamey** [comment]
I'd like to use Deskreen to replace Discord gameplay streaming on Linux, but without audio streaming, it's just as lacking as Discord on Linux is, so there's not much of a point. Other than this, it works great! Even online, all I have to do is port forward Deskreen in my router, allow that port through the firewall, and give them my public IP address instead of my local IP, and my friends can nicely watch my gameplay. Something nice would be to support TLS with https:// connections, so that my stream is secured.

**Comment by davidtomw** [comment]
Deskreen is fantastic. Thank you. My business offers photography workshops on all seven continents. An essential part of our instruction happens during our image reviews and processing sessions. Before Deskreen, we relied on poorly calibrated TVs in meeting rooms or had to bring a projector with even worse color accuracy. With Deskreen, we can share our screens with all the clients on their individual laptops. With a battery power supply for a travel router, we can even do this while camping in the wilderness. We can have an impromptu meeting room with shared content anywhere we travel. This is such a big improvement. 

I don't know if this is possible, but I would gladly pay someone to create a custom-branded version, so that our clients see our branding when they connect.

Also, we have to connect up to 8-12 people at a time, and the current process of them joining one-by-one is a bit tedious.

**Reply by pavlobu** [reply]
Hi @davidtomw! I’m beyond excited to hear how Deskreen is transforming your workshops! As the repo owner, I’d love to craft a custom-branded version tailored just for you, ensuring it aligns perfectly with your vision.

Working directly with me makes the process smooth and hassle-free, as it avoids any AGPL-related complexities that come if someone else modifies Deskreen’s code, since they’d need to share their changes publicly under the AGPL license.

Let’s create something awesome together—please reach out at pavlobu@gmail.com to chat!

**Comment by iannz-wai** [comment]
Agree with all the comments about Deskreen being fantastic!
What I would like to use it for is being able to mirror devices to the open source white board program OpenBoard (https://openboard.ch/).
While I can use the builtin web browser or an external web browser to mirror a screen as soon as I try to embed that window onto the 'whiteboard' it disconnects. I have tried this on Windows 11 and Ubuntu 24.04 (both as both sender and receiver).

Steps in OpenBoard:
* Click Web (top right corner)
* Enter the URL from Deskreen and follow the usual Deskreen instructions for mirroring (works like a charm :) )
* Click the top icon on the right hand floating toolbar for creating the embedded webpage (whiteboard calls it an application)
At this point the enclosed browser window shows 'Redirecting to target' and then the Deskreen Error Dialog  'You were not allowed to Connect'). And thus I cannot get it to have a window of the mirrored device on the whiteboard.

Is this is a deliberate security measure by Deskreen?

Would be great to have these two open source programs work together. I'm about to post a similar message to OpenBoard as I'm not sure where the problem lies between the two programs.

However, I can use PlainApp (https://docs.plain.icu/) for Android devices like this in OpenBoard and it works.

Any suggestions would be greatly appreciated.

**Reply by iannz-wai** [reply]
I have found a work around solution:
https://github.com/OpenBoard-org/OpenBoard/discussions/1334

**Comment by alex-larkin** [comment]
We conduct workshops six times a year in which a consultant connects remotely to a conference room in which about nine people view a projector screen, where the consultant is sharing his screen. However, due to viewing distance he must use a large font size, limiting the amount of data that can fit on the screen. 

We want to try using Deskscreen in this way: the consultant shares his screen to the local team lead, and then the team lead uses Deskscreen to share that same screen to the other laptops on the local wifi network. 

We understand that with the pro feature it will be possible for multiple machines to connect to the Deskscreen "livestream" and for each of of those devices to save the URL as a favorite. 

Beyond being able to increase the data visible on the screen by using a smaller font size, it will also be reassuring to have two methods for collaboration (projector and Deskscreen).

Any idea on how much the pro edition will cost, and when it's coming out?

Thanks so much!

**Comment by pavlobu** [comment]
@alex-larkin 
Hi Alex,

Thank you for your interest in Deskreen!

The app will come with different tiered versions offering various features and pricing options to suit different user needs. We are still analyzing the market and figuring out the right pricing model.

The premium version app is planned to be released sometime around the end of January or the beginning of February this year.

Stay tuned!

**Reply by alex-larkin** [reply]
Thanks @pavlobu for the info! Looking forward to the next release.

**Reply by pavlobu** [reply]
Hi @alex-larkin 

I'm pleased to inform you that the release of Deskreen Pro and Deskreen Teams has been successful.
 
You can download them here: https://deskreen.com/download/
 
Have a great day!
